src/revivalpmmp/pureentities/entity/other/XPOrb.php

```diff
@@ -23,99 +23,106 @@
 use pocketmine\Player;
 use revivalpmmp\pureentities\PluginConfiguration;
 use revivalpmmp\pureentities\sound\ExpPickupSound;
+use pocketmine\math\Vector3;
 
 class XPOrb extends Entity {
-	const NETWORK_ID = 69;
+    const NETWORK_ID = 69;
 
 	public $width = 0.25;
-	public $length = 0.25;
 	public $height = 0.25;
 
-	protected $gravity = 0.04;
-	protected $drag = 0;
-	
-	protected $experience = 0;
-	
-	protected $range = 6;
+    protected $gravity = 0.04;
+    protected $drag = 0;
 
-	public function initEntity(){
-		parent::initEntity();
+    protected $experience = 0;
+
+    protected $range = 6;
+
+    public function initEntity(){
+        parent::initEntity();
         if (PluginConfiguration::getInstance()->getEnableNBT()) {
             if (isset($this->namedtag->Experience)) {
                 $this->experience = $this->namedtag["Experience"];
             } else $this->close();
         }
-	}
-
-	public function onUpdate(int $currentTick): bool {
-		if($this->closed){
-			return false;
-		}
-		
-		$tickDiff = $currentTick - $this->lastUpdate;
-		
-		$this->lastUpdate = $currentTick;
-		
-		$this->timings->startTiming();
-		
-		$hasUpdate = $this->entityBaseTick($tickDiff);
-
-		$this->age++;
-
-		if($this->age > 1200){
-			$this->kill();
-			$this->close();
-			$hasUpdate = true;
-		}
-		
-		$minDistance = PHP_INT_MAX;
-		$target = null;
-		foreach($this->getViewers() as $p){
-			if(!$p->isSpectator() and $p->isAlive()){
-				if(($dist = $p->distance($this)) < $minDistance and $dist < $this->range){
-					$target = $p;
-					$minDistance = $dist;
-				}
-			} 
-		}
-		
-		if($target !== null){
-			$moveSpeed = 0.7;
-			$motX = ($target->getX() - $this->x) / 8;
-			$motY = ($target->getY() + $target->getEyeHeight() - $this->y) / 8;
-			$motZ = ($target->getZ() - $this->z) / 8;
-			$motSqrt = sqrt($motX * $motX + $motY * $motY + $motZ * $motZ);
-			$motC = 1 - $motSqrt;
-		
-			if($motC > 0){
-				$motC *= $motC;
-				$this->motionX = $motX / $motSqrt * $motC * $moveSpeed;
-				$this->motionY = $motY / $motSqrt * $motC * $moveSpeed;
-				$this->motionZ = $motZ / $motSqrt * $motC * $moveSpeed;
-			}
-
-			$this->motionY -= $this->gravity;
-
-			if($this->checkObstruction($this->x, $this->y, $this->z)){
-				$hasUpdate = true;
-			}
-
-			if($this->isInsideOfSolid()){
-				$this->setPosition($target);
-			}
-
-			if($minDistance <= 1.3){
+    }
+
+    public function onUpdate(int $currentTick): bool {
+        if($this->isClosed()){
+            return false;
```

Comment:
Use tabs!

Reply:
Ok, I often forget to check my IDE settings every time I commit to the 30 or so plugins to which I contribute. Why we can't all adopt the PSR-2 convention for this is beyond me.

Reply:
PocketMine uses tabs, so I figured I might as well do the same 😋